xcbgen-rs/src/defs.rs

```diff
@@ -705,11 +1140,17 @@ impl VariableSize {
         }
     }
 
+    /// Create an instance that describes a zero-sized type.
     #[inline]
     pub fn zero() -> Self {
         Self { base: 0, incr: 0 }
     }
 
+    /// Return a description of the size of things when `self` is appended to `other`.
+    // FIXME: This code makes no sense to me. When I have something of size `1+4*n` and something
+    // of size `2+8*m`, the result has sizes `3+4*n+8*m` and not `3+4*l`. There is a discrepancy
+    // between what the docs for `VariableSize` say what it represents and what this method is
+    // supposed to do.
```

Comment:
The reasoning here is:
`3 + 4*n + 8*m = 3 + 4*n+4*2*m = 3 + 4*(n + 2*m) = 3 + 4*l`

Reply:
Note that `VariableSize` might represent a superset of the actual sizes of the object.

Reply:
Okay, I went with the original comment that explained its meaning and it sounded like it was describing exact sizes. I guess I accidentally figured out that this needs better docs. :-)

My (current) plan (which can of course still change) for next weekend is to finish adding some doc comments (at least the parts that I understood) and then to split up the file a bit into smaller parts. Afterwards, this alignment calculations get some unit tests.

Reply:
What do you think of https://github.com/psychon/x11rb/pull/397/commits/b5540185fcfd46c330644f9b7d7fa78c05d1c776 ?